runtime/lua/vim/diagnostic.lua

```diff
@@ -1535,6 +1535,10 @@ M.handlers.signs = {
       return
     end
 
+    if not opts.signs then
+      return
+    end
```

Comment:
don't think this change is needed? the line just below checks `opts.signs and ...` which suggests that nil `opts.signs` is expected and supported.

your other changes LGTM